### ISSUE TITLE
OSD / Mirador issues

### ISSUE THREAD

**Issue by eroux** [buda-base/public-digital-library#48]
As a context: for images that are originally tiffs, we want to use png, not jpg. For this we introduced a `"formatHints": ["png", "jpg"]` in the services ([example](http://iiif.bdrc.io/bdr:V22084_I0886::08860004.tif/info.json)) and also in the manifests ([example](http://presentation.bdrc.io/2.1.1/v:bdr:V22084_I0886/manifest)).

We also have an optimization on the server side (only for jpg but let's not keep that distinction here) where images requested in their original form are much faster to serve than images that need to be processed. On the server we detect these images with their url form: `/full/full/0/default.jpg` or `/full/max/0/default.jpg`. On the other hand, equivalent images that use the maximum width are longer to process. So for instance if an image has a width of `1234` and a height of `567`, the two equivalent requests
- `/full/max/0/default.jpg`
- `/0,0,1234,567/1234,/0/default.jpg`

will have a very different processing time, the first being much faster.  The first one is called the canonical form. (it's actually not really true but it will in v3).

There is already some code to leverage that in OSD (see [this issue](https://github.com/openseadragon/openseadragon/issues/1617), and also in Mirador (see [this commit](https://github.com/buda-base/mirador/commit/9b8c19ca10d1bf327c3ea8c50368dd12460afd28)). But for some reason, some of the URLs actually requested by the current system are still in the form `/full/1234,/...` instead of `/full/max/...`, I don't know where this comes from.

Note that Mirador has its own URL building for thumbnails. Basically we want Mirador to only request the max image for thumbnail (not a reduce size). The trick we use for that is that the manifests advertise a level0 compliance, which should prevent resizing (see [this issue](https://github.com/ProjectMirador/mirador/issues/2153) and [this commit](https://github.com/buda-base/mirador/commit/9b8c19ca10d1bf327c3ea8c50368dd12460afd28)).

So in short we need to make sure that:
- [x] level0 is respected in Mirador and thumbnails are always requested at the max dimension
- [x] pngs are requested in thumbnails by Mirador
- [x] pngs are requested by OpenSeaDragon in the image view (when viewing one image only)
- [x] when a full thumbnail is requested (at max width) by mirador, it is requested using the canonical url
- [x] when an image is requested (at max width) by openseadragon, it is requested using the canonical url

**Comment by berger-n** [comment]
I think we are not far from all done here now: http://library-dev.bdrc.io/show/bdr:W22084
(tell me whether it is normal or not that http://library-dev.bdrc.io/show/bdr:W22073 is still using jpg)

**Comment by eroux** [comment]
Perfect, thanks a lot! it's really a major performance improvement
